### PR TITLE
Added WorkerRegistry and ability to add additional worker implementat…

### PR DESCRIPTION
**Pull request by iborysenko**
…ions (grpc, jobs, ...)

### CODE REVIEW

**Comment by iborysenko** [comment]
Hi @Baldinof 
This is preparation PR for future roadrunner/jobs integration 

Now it will be possible to add something like 
```php

final class JobsWorker implements WorkerInterface
{
    public function __construct(
        private EventDispatcherInterface $eventDispatcher,
        private ConsumerInterface $consumer,
        private ProcessorRegistry $registry,
        private ChainExceptionHandler $exceptionHandler
    ) {
    }

    public function start(): void
    {
        $this->eventDispatcher->dispatch(new WorkerStartEvent());

        while ($task = $this->consumer->waitTask()) {
            $this->eventDispatcher->dispatch(new TaskReceivedEvent($task));
            $exception = null;
            try {
                $processor = $this->registry->getProcessor($task->getName());
                if (!$processor instanceof ProcessorInterface) {
                    throw new JobsException(\sprintf('missing processor for task "%s"', $task->getName()));
                }

                $task = $processor->process($task);
            } catch (\Throwable $e) {
                $exception = $e;
                $task = $this->exceptionHandler->handle($exception, $task);
            } finally {
                if (false === $task->isCompleted()) {
                    $task->complete();
                }
                $this->eventDispatcher->dispatch(new TaskCompletedEvent($task, $exception));
            }
        }

        $this->eventDispatcher->dispatch(new WorkerStopEvent());
    }
}

```

**Comment by Baldinof** [comment]
Hi!

Nice, I'll try to review it this week :)

File: src/Worker/GrpcWorkerInterface.php [hunk 1]
```diff
@@ -1,10 +0,0 @@
-<?php
-
-declare(strict_types=1);
-
-namespace Baldinof\RoadRunnerBundle\Worker;
```
Comment (Baldinof) [inline src/Worker/GrpcWorkerInterface.php:5]: If we remove an interface, that would mean a 3.0 release. Maybe you can just make the `GrpcWorkerInterface` extends the `WorkerInterface`, and deprecate `GrpcWorkerInterface`.

So we keep backward compatibility.

File: src/Worker/WorkerRegistry.php [hunk 2]
```diff
@@ -0,0 +1,19 @@
+<?php
+declare(strict_types=1);
+
+namespace Baldinof\RoadRunnerBundle\Worker;
+
+class WorkerRegistry implements WorkerRegistryInterface
```
Comment (Baldinof) [inline src/Worker/WorkerRegistry.php]:
```suggestion
/**
 * @internal
 */
final class WorkerRegistry implements WorkerRegistryInterface
```

**Review by Baldinof — COMMENTED** [review]
Some minor comments. Great work :)

File: src/Worker/GrpcWorkerInterface.php (hunk 1 above)
Comment (iborysenko) [reply src/Worker/GrpcWorkerInterface.php:5]: added deprecation

**Comment by FluffyDiscord** [comment]
@iborysenko would you mind adding full usage example to README ?

**Comment by iborysenko** [comment]
> FluffyDiscord

Hi, you meant full roadrunner-jobs configuration, or symfony flex setup ? 
My implementation of roadrunner-jobs will be in the next MR, this MR is just structure preparation

**Comment by FluffyDiscord** [comment]
> > FluffyDiscord
> 
> Hi, you meant full roadrunner-jobs configuration, or symfony flex setup ? My implementation of roadrunner-jobs will be in the next MR, this MR is just structure preparation

Aha, then I am looking forward to the next MR 😊

**Review by Baldinof — APPROVED** [review]

**Comment by Baldinof** [comment]
Thank you @iborysenko!

**Comment by koekaverna** [comment]
It's awesome!
Waiting for release!

**Pull request merged**